File: frontend/src/types/agendamento.ts
export type AgendamentoStatus = "AGENDADO" | "CONCLUIDO" | "CANCELADO";

export interface Agendamento {
  id: number;
  clienteId: number;
  clienteNome: string;
  veiculoId: number;
  veiculoPlaca: string;
  dataHora: string;
  descricao: string;
  status: AgendamentoStatus;
}

export interface AgendamentoPayload {
  clienteId: number;
  veiculoId: number;
  dataHora: string;
  descricao: string;
  status: AgendamentoStatus;
}

File: src/main/java/com/jhonycar/backend/entity/Agendamento.java
package com.jhonycar.backend.entity;

import com.jhonycar.backend.entity.enums.AgendamentoStatus;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "agendamentos")
public class Agendamento {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "cliente_id", nullable = false)
    private Cliente cliente;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "veiculo_id", nullable = false)
    private Veiculo veiculo;

    @Column(nullable = false)
    private LocalDateTime dataHora;

    @Column(nullable = false)
    private String descricao;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private AgendamentoStatus status;

    @PrePersist
    public void prePersist() {
        if (status == null) {
            status = AgendamentoStatus.AGENDADO;
        }
    }
}
